app/Http/Controllers/Order/ExtendedOrderController.php

```diff
@@ -116,6 +118,62 @@ public function allInstallations($allInstallation,$join)
         return $join;
     }
 
+    /**
+     * Searches for InActive Installation
+     *
+     * @param  string $allInstallation
+     * @param  App\Model\Order $join The order instance
+     *
+     * @return $join
+     */
+    public function allInactiveInstallations($allInstallation,$join)
+    {
+        if($allInstallation) {
+            $dayUtc = new Carbon('-30 days');
+            $minus30Day = $dayUtc->toDateTimeString();
+            if($allInstallation == 'paid_inactive_ins') {
```

Comment:
This entire logic can be simplified to 
```
$baseQuery = $join->whereHas('subscription', function($query) use($minus30Day) {
                    $query->where('updated_at', '<', $minus30Day);

return $baseQuery->when($allInstallation = 'paid_inactive_ins', function($q){
		$q->where('price_override', '>', 0)
	})->when($allInstallation == 'unpaid_inactive_ins', function($q){
		$q->where('price_override', '=', 0)
	})

```


for these kind of scenarios, we can use a when query

Reply:
Fixed